### PR TITLE
[Find Forms] Incorporate new validPdf flag

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
The vets-api PR below added a new flag into the Forms API method called `valid_pdf`. This indicates whether a form has been confirmed to have a valid PDF download. If there isn't, we shouldn't show it.

Todo
- [x] Update the `find-forms/constants/stub.json` with the new response JSON from `dev-api/v0/forms` to include the `valid_pdf` attribute. Also may need to update the E2E test to filter out invalid forms.

https://github.com/department-of-veterans-affairs/vets-api/pull/4224
https://github.com/department-of-veterans-affairs/va.gov-team/issues/8648
## Testing done
- [x] Confirm forms missing on `/vaforms/` have `valid_pdf: false` and are in return not rendered in this React widget
- [x] Confirm forms that link to HTML pages still come through in this React widget

## Acceptance criteria
- [ ] Stale forms are no longer shown in search results

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/tests/00-required.e2e.spec.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ function runTest(browser) {
   );
   browser.axeCheck(SELECTORS.WIDGET);
 
-  const sortedForms = sortBy(stub.data, 'id');
+  const validForms = stub.data.filter(form => form.attributes.validPdf);
```
Comment (kelsonic) [inline src/applications/find-forms/tests/00-required.e2e.spec.js:36]: Just confirming here, but it looks like `stub.data` has changed from an Object to an Array, is that correct/intended? 🙂Will `stub.data` ever be `undefined`? If so we may want to do some `?.` here

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: src/applications/find-forms/tests/00-required.e2e.spec.js (hunk 1 above)
Comment (ncksllvn) [reply src/applications/find-forms/tests/00-required.e2e.spec.js:36]: I think `.data` was an array before but it was written kinda cleverly in the `sortBy`. Am I reading it wrong? (not a _ expert 🙂 )

It's a static JSON blob so I'd like the e2e test fail if something is missing

Comment (kelsonic) [reply src/applications/find-forms/tests/00-required.e2e.spec.js:36]: Oh you're right! It's always been an array, I just read it wrong 🤪

**Pull request merged**